packages/remix-dev/__tests__/init-test.ts

```diff
@@ -31,6 +32,31 @@ jest.mock("child_process", () => {
   };
 });
 
+// Copy over the current build of @remix-run/dev and execute remix init from the node_modules dir there.
+// Certain bugs only manifest when using this approach.
+const execRemixInitInProject = (projectDir: string) => {
```

Comment:
This stuff with copying over the newly-built version of @remix-run/dev and running it as a node module in the temp project is clunky, but was the only way I could reproduce the failing behavior (es6 module init script erroring) in a unit test. 

It seems like some aspect of the `run(["init"])` calls used by other tests here doesn't fully replicate the real-world behavior of running `remix init` as a project dependency, so this was the only workaround I found initially